lib/mcb/courses_editor.rb

```diff
@@ -21,24 +21,46 @@ def initialize(provider:, requester:, course_codes: [])
 
     def run
       finished = false
-      puts "Editing #{course_codes.join(', ')}"
-      print_at_most_two_courses
       until finished
-        choice = @cli.main_loop
+        puts "Editing #{course_codes.join(', ')}"
+        print_at_most_two_courses
+        choice = main_loop
 
-        if choice.start_with?("edit")
+        if choice.nil?
+          finished = true
+        elsif choice == 'edit subjects'
+          edit_subjects
+        elsif choice.start_with?("edit")
           attribute = choice.gsub("edit ", "").gsub(" ", "_").to_sym
           edit(attribute)
         elsif choice =~ /sync .* to Find/
           sync_courses_to_find
-        elsif choice == "exit"
-          finished = true
         end
       end
     end
 
   private
 
+    def main_loop
+      choices = [
+        "edit title",
+        "edit course code",
+        "edit maths",
+        "edit english",
+        "edit science",
+        "edit route",
+        "edit qualifications",
+        "edit study mode",
+        "edit accredited body",
+        "edit start date",
+        "edit application opening date",
+        "edit age range",
+        "edit subjects",
+        "sync course(s) to Find"
+      ].grep_v(->(c) { c.in?(["edit subjects"]) && @courses.count != 1 })
```

Comment:
I can see what this does (having read https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-grep_v ) but I'm not clear from the code why it's needed.

Reply:
I've spun out the `filter_single_course_options_if_necessary` which should hopefully make things clearer.